pennylane/ops/functions/equal.py

```diff
@@ -164,23 +167,32 @@ def _equal_measurements(op1: MeasurementProcess, op2: MeasurementProcess, **kwar
         observables_match = op1.obs == op2.obs
     wires_match = op1.wires == op2.wires
     eigvals_match = qml.math.allequal(op1.eigvals(), op2.eigvals())
-    log_base_match = getattr(op1, "log_base", None) == getattr(op2, "log_base", None)
-
-    return (
-        return_types_match
-        and observables_match
-        and wires_match
-        and eigvals_match
-        and log_base_match
-    )
+
+    return return_types_match and observables_match and wires_match and eigvals_match
 
 
 @_equal.register
 # pylint: disable=unused-argument
-def _equal_shadow_measurements(
-    op1: ShadowMeasurementProcess, op2: ShadowMeasurementProcess, **kwargs
-):
-    """Determine whether two ShadowMeasurementProcess objects are equal"""
+def _(op1: _VnEntropy, op2: _VnEntropy, **kwargs):
+    """Determine whether two MeasurementProcess objects are equal"""
+    eq_m = _equal_measurements(op1, op2)
+    log_base_match = op1.log_base == op2.log_base
+    return eq_m and log_base_match
+
+
+@_equal.register
+# pylint: disable=unused-argument
+def _(op1: _MutualInfo, op2: _MutualInfo, **kwargs):
+    """Determine whether two MeasurementProcess objects are equal"""
+    eq_m = _equal_measurements(op1, op2)
+    log_base_match = op1.log_base == op2.log_base
+    return eq_m and log_base_match
+
+
+@_equal.register
+# pylint: disable=unused-argument
+def _equal_shadow_measurements(op1: _ShadowExpval, op2: _ShadowExpval, **kwargs):
+    """Determine whether two ClassicalShadow objects are equal"""
```

Comment:
Do we need one for `ClassicalShadow` as well?

Reply:
The `ClassicalShadow` class only has one extra argument that the `MeasurementProcess` class, which is `self.seed`. However the seed wasn't used in `qml.equal`, thus we can use the `MeasurementProcess` branch to test it.